tests/test_happy_path.py

```diff
@@ -0,0 +1,192 @@
+    farming_rewards.addGift(ldo_token, rewards_amount, rewards_manager, scale, {"from": stranger})
+
+    # DAO transfers LDO tokens to reward manager
+    assert ldo_token.balanceOf(dao_agent) >= rewards_amount
+
+    transfer_calldata = ldo_token.transfer.encode_input(rewards_manager, rewards_amount)
+
+    dao_agent.execute(
+        ldo_token, 0, transfer_calldata, {"from": dao_voting_impersonated}
+    )
+
+    assert ldo_token.balanceOf(rewards_manager) == rewards_amount
+
+    # 1INCH distributor transfers tokens to farming contract directly
+    one_inch_token.transfer(farming_rewards, rewards_amount, {"from": one_inch_token_owner})
+
+    assert one_inch_token.balanceOf(farming_rewards) == rewards_amount
+
+    # Someone starts new reward period
+    # 1INCH via their distributor
+    farming_rewards.notifyRewardAmount(0, rewards_amount, {"from": stranger})
+
+    # LDO via reward manager
+    rewards_manager.start_next_rewards_period({"from": stranger})
+
+    assert rewards_manager.is_rewards_period_finished() == False
+
+    # Users A and B are obtaining stETH and DAI tokens somehow
+    user_a, user_b = accounts[2], accounts[3]
+
+    user_a.transfer(steth_token, Wei("3 ether"))
+    user_b.transfer(steth_token, Wei("3 ether"))
+
+    assert steth_token.balanceOf(user_a) > 0
+    assert steth_token.balanceOf(user_b) > 0
+
+    dai_token.transfer(user_a, Wei("3 ether"), {"from": dai_holder})
+    dai_token.transfer(user_b, Wei("3 ether"), {"from": dai_holder})
+
+    assert dai_token.balanceOf(user_a) > 0
+    assert dai_token.balanceOf(user_b) > 0
+
+    # Users A and B deposit their stETH and DAI tokens to liquidity pool to receive LP tokens
+    steth_token.approve(mooniswap, 1500, {"from": user_a})
+    dai_token.approve(mooniswap, 1500, {"from": user_a})
+
+    steth_token.approve(mooniswap, 2000, {"from": user_b})
+    dai_token.approve(mooniswap, 2000, {"from": user_b})
+
+    mooniswap.deposit([1500, 1500], [1000, 1000], {"from": user_a})
+    mooniswap.deposit([2000, 2000], [1500, 1500], {"from": user_b})
+
+    user_a_balance_lp = mooniswap.balanceOf(user_a)
+    user_b_balance_lp = mooniswap.balanceOf(user_b)
+
+    assert user_a_balance_lp > 0
+    assert user_b_balance_lp > 0
+
+    mooniswap.approve(farming_rewards, user_a_balance_lp, {"from": user_a})
+    mooniswap.approve(farming_rewards, user_b_balance_lp, {"from": user_b})
+
+    # Users A and B are staking their LP tokens to reward farming contract
+    farming_rewards.stake(user_a_balance_lp, {"from": user_a})
+    farming_rewards.stake(user_b_balance_lp, {"from": user_b})
+
+    assert farming_rewards.balanceOf(user_a) > 0
+    assert farming_rewards.balanceOf(user_b) > 0
+
+    # User A exists from reward farming contract after half of rewarding period
+    chain.sleep(rewards_period // 2)
+    chain.mine()
+
+    # TODO: This is not working for some weird reason. Investigate.
+    # assert rewards_manager.is_rewards_period_finished() == False
```

Comment:
Still, prefer to clarify this.

Reply:
Fixed in https://github.com/maddevsio/lido/pull/3/commits/9faeb9f59db243367422c0632694b23db3fdaff8 and https://github.com/maddevsio/lido/pull/3/commits/e0ab543202a3d080a40b2ef7da8e7c5e50eea450. @pavelis please note these fixes also. Just in case.